File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProduct/IfcElementType/IfcBuiltElementType/IfcRoofType/IfcRoofType.vb
Public Class IfcRoofType Inherits IfcBuiltElementType
    Public PredefinedType As IfcRoofTypeEnum

    ' === WHERE CLAUSES ===
    ' CorrectPredefinedType : (PredefinedType <> IfcRoofTypeEnum.USERDEFINED) OR ((PredefinedType = IfcRoofTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType))
End Class

File: src/main/resources/output/IFC4X3_java/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProduct/IfcElementType/IfcBuiltElementType/IfcRoofType/IfcRoofType.java
public class IfcRoofType extends IfcBuiltElementType {
    public IfcRoofTypeEnum PredefinedType;

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : (PredefinedType <> IfcRoofTypeEnum.USERDEFINED) OR ((PredefinedType = IfcRoofTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType))
}
